### ISSUE TITLE
rdflib.collections issues

### ISSUE THREAD

**Issue by gromgull** [RDFLib/rdflib#223]
Adding the same element twice to a collection triggers an exception when serializing the graph, but this shouldn't be a problem:

```
/usr/local/lib/python2.7/dist-packages/rdflib/collection.pyc in __len__(self)
     77         links=set()
     78         for item in self.graph.items(self.uri):
---> 79             assert item not in links,"There is a loop in the RDF list! (%s has been processed before)"%item
     80             links.add(item)
     81             count += 1

AssertionError: There is a loop in the RDF list! (http://www.w3.org/2000/01/rdf-schema#Class has been processed before)
```

There is also no way to close the list, appending RDF.nil adds another bnode with first->nil and rest->nil.

**Comment by wwaites** [comment]
This one just bit me

**Comment by gromgull** [comment]
@wwaites if you find a fix in the next weeks we can get it into 4.2.2!

**Comment by wwaites** [comment]
Near as I can tell, it is safe to remove this assertion. Graph.items [checks for recursion](https://github.com/RDFLib/rdflib/blob/53258d3a57e9c4b22fac9372ff3dbaa6296b8df7/rdflib/graph.py#L801) on the list structure itself . This check is simply incorrect for [rdf:List](https://www.w3.org/TR/2004/REC-rdf-schema-20040210/#ch_collectionvocab) that is quite unrestricted. There is no type that I know of with set semantics so I can't see where the assertion can ever be correct even for containers.
